src/operator/numpy/np_matrix_op.cc

```diff
@@ -1325,5 +1326,27 @@ NNVM_REGISTER_OP(_backward_np_diag)
 .set_attr<nnvm::TIsBackward>("TIsBackward", true)
 .set_attr<FCompute>("FCompute<cpu>", NumpyDiagOpBackward<cpu>);
 
+NNVM_REGISTER_OP(_npi_diagflat)
```

Comment:
No need to make this an `_npi_` operator, simply use `_np_` and add docs in python/mxnet/_numpy_op_doc.py after `_np_diag` in the file.